### ISSUE TITLE
[CustomSpawns] Primary spawns reverting on round start

### ISSUE THREAD

**Issue by Turbine1991** [GhoulofGSG9/ShineExtraPlugins#2]
Here's the minimal steps to reproduce this issue by yourself:

Step 1
Use this extremely minimal custom spawn for "ns2_biodome.json".

[{
  "enemyspawns":["Platform"],
  "name":"Hydroponics",
  "team":"marines"
}]
Step 2
Load ns2_biodome
Join a team. (At this point they appear to work)

Step 3
Start a game using either "sv_cheats 1" - followed by "sv_cheats 1" or restart the pre-game spawn locations using "sv_reset".

Outcome
Marines will always spawn in Reception whenever the round starts or spawns are subsequently chosen. Interestingly the enemy spawns continue to work fine for aliens, based off the desired marine spawn locations.

Expected Outcome
Marines to always spawn in Hydroponics and not Reception after every spawn reset.

**Comment by GhoulofGSG9** [comment]
Should be fixed with https://github.com/GhoulofGSG9/ShineExtraPlugins/commit/ec7e591effeb67bc19ab01eda38d414ffbe591ef

**Comment by Turbine1991** [comment]
Thank you. While there's still a few bugs, it's at least workable:

**Bugs/Errors**
-When no enemyspawns are provided, it throws an error. Even for a duplicate entry for the opposite team.
-All generated map configs now throw an arrow as they are all missing the new requirement of enemy spawns.
-The default config generated for ns2_kodiak has a spelling mistake, throwing an error due to the word "Asteroid" being spelt as "Astroid".

**Working configs**

ns2_veil:

```
[{
  "enemyspawns":["Control"],
  "name":"Pipeline",
  "team":"aliens"
},{
  "enemyspawns":["Control", "Pipeline"],
  "name":"Sub-Sector",
  "team":"aliens"
},{
  "enemyspawns":["Sub-Sector", "Pipeline"],
  "name":"Control",
  "team":"marines"
},{
  "enemyspawns":["Sub-Sector"],
  "name":"Pipeline",
  "team":"marines"
}]
```

ns2_eclipse:

```
[{
  "enemyspawns":["Maintenance"],
  "name":"Eclipse",
  "team":"both"
},{
  "enemyspawns":["Eclipse"],
  "name":"Maintenance",
  "team":"both"
}]
```

ns2_kodiak:

```
[{
  "enemyspawns":["Hangar Bay", "Command"],
  "name":"Asteroid Tracking",
  "team":"marines"
},{
  "enemyspawns":["Asteroid Tracking"],
  "name":"Hangar Bay",
  "team":"aliens"
},{
  "enemyspawns":["Hangar Bay"],
  "name":"Surface Access",
  "team":"marines"
},{
  "enemyspawns":["Asteroid Tracking"],
  "name":"Command",
  "team":"aliens"
}]
```

**Comment by GhoulofGSG9** [comment]
Thanks for reporting those things. 

The enemyspawns entry check gets now ignored for alien team spawns. I also fixed the typo in the ns2_kodiak default map config.

Furthermore I added a new command to the plugin you might find useful: 

`sh_dumpspawns` will dump the vanilla spawn set-up of the current map into a map-config named `<mapname>_dumped.json`. That file might help out as template to create own set-ups.
